Analytics configuration is the responsibility of the host application when running in widget mode (#3089)
vector-im/element-call: docs/url-params.md

| `viaServers`                                            | Comma separated list of [Matrix Server Names](https://spec.matrix.org/v1.12/appendices/#server-name) | Not applicable                                                                                    | No                           | Homeserver for joining a room, non-empty value required for rooms not on the user’s default homeserver.                                                                                                                                                                                                                                                                                             |

### Widget-only parameters

> These parameters are only available in [widget](./embedded-standalone.md) mode.

Author: ```suggestion
These parameters are only supported in [widget](./embedded-standalone.md) mode.
```

| `userId`        | [Matrix User Identifier](https://spec.matrix.org/v1.12/appendices/#user-identifiers)      | Yes                     | The Matrix user ID.                                                                                                                                                                                                                                                                                                                                |
| `widgetId`      | [MSC2774](https://github.com/matrix-org/matrix-spec-proposals/pull/2774) format widget ID | Yes                     | The id used by the widget. The presence of this parameter implies that element call will not connect to a homeserver directly and instead tries to establish postMessage communication via the `parentUrl`.                                                                                                                                        |

### Embedded-only parameters

> These parameters are only available in the [embedded](./embedded-standalone.md) package of Element Call and will be ignored in the [full](./embedded-standalone.md) package.

Author: ```suggestion
These parameters are only supported in the [embedded](./embedded-standalone.md) package of Element Call and will be ignored in the [full](./embedded-standalone.md) package.
```

| `deviceId`           | Matrix device ID                                                                          | Yes                                 | The Matrix device ID for the widget host.                                                                                                                                                                                                                                                                                                          |
| `parentUrl`          |                                                                                           | Yes                                 | The url used to send widget action postMessages. This should be the domain of the client or the webview the widget is hosted in. (in case the widget is not in an Iframe but in a dedicated webview we send the postMessages same WebView the widget lives in. Filtering is done in the widget so it ignores the messages it receives from itself) |

> | `posthogApiHost`     | Posthog server URL                                                                        | No                                  | e.g. `https://posthog-element-call.element.io`                                                                                                                                                                                                                                                                                                     |

Author: ```suggestion
| `posthogApiHost`     | Posthog server URL                                                                        | No                                  | e.g. `https://posthog-element-call.element.io`. n.b. the value set in config is never used when running as widget                                                                                                                                                                                                                                                                                                     |
```

> | `posthogApiKey`      | Posthog project API key                                                                   | No                                  |                                                                                                                                                                                                                                                                                                                                                    |

Author: ```suggestion
| `posthogApiKey`      | Posthog project API key                                                                   | No                                  | n.b. the value set in config is never used when running as widget                                                                                                                                                                                                                                                                                                                                                   |
```

| `rageshakeSubmitUrl` | Rageshake server URL endpoint                                                             | No                                  | e.g. `https://element.io/bugreports/submit`                                                                                                                                                                                                                                                                                                        |
| `returnToLobby`      | `true` or `false`                                                                         | No, defaults to `false`             | Displays the lobby in widget mode after leaving a call; shows a blank page if set to `false`. Useful for video rooms.                                                                                                                                                                                                                              |

> | `sentryDsn`          | Sentry [DSN](https://docs.sentry.io/concepts/key-terms/dsn-explainer/)                    | No, defaults to the value in config |                                                                                                                                                                                                                                                                                                                                                    |

Author: ```suggestion
| `sentryDsn`          | Sentry [DSN](https://docs.sentry.io/concepts/key-terms/dsn-explainer/)                    | No | n.b. the value set in config is never used when running as widget                                                                                                                                                                                                                                                                                                                                                   |
```

> | `sentryEnvironment`  | Sentry [environment](https://docs.sentry.io/concepts/key-terms/key-terms/)                | No, defaults to the value in config |                                                                                                                                                                                                                                                                                                                                                    |

Author: ```suggestion
| `sentryEnvironment`  | Sentry [environment](https://docs.sentry.io/concepts/key-terms/key-terms/)                | No | n.b. the value set in config is never used when running as widget                                                                                                                                                                                                                                                                                                                                                   |
```
